libc/utils/gpu/server/rpc_server.cpp

```diff
@@ -206,6 +206,18 @@ void handle_printf(rpc::Server::Port &port) {
     delete[] reinterpret_cast<char *>(ptr);
 }
 
+namespace {
+struct TempStorage {
+  char *alloc(size_t size) {
+    storage.emplace_back(std::make_unique<char[]>(size));
+    return storage.back().get();
+  }
+
+  // TODO: SmallVector
```

Comment:
```suggestion
```

Reply:
If this were LLVM we could probably just use a bump pointer, but this is good enough.